haystack/nodes/query_classifier/transformers.py

```diff
@@ -57,43 +61,75 @@ class TransformersQueryClassifier(BaseQueryClassifier):
         batch_size: Optional[int] = None,
     ):
         """
-        :param model_name_or_path: Transformer based fine tuned mini bert model for query classification
+        :param model_name_or_path: Directory of a saved model or the name of a public model, for example 'shahrukhx01/bert-mini-finetune-question-detection'.
+        See [Hugging Face models](https://huggingface.co/models) for a full list of available models.
+        :param model_version: The version of the model to use from the Hugging Face model hub. This can be a tag name, a branch name, or a commit hash.
+        :param tokenizer: The name of the tokenizer (usually the same as model).
         :param use_gpu: Whether to use GPU (if available).
+        :param task: Specifies the type of classification. Possible values: 'text-classification' or 'zero-shot-classification'.
+        :param labels: If the task is 'text-classification' and an ordered list of labels is provided, the first label corresponds to output_1,
+        the second label to output_2, and so on. The labels must match the model labels; only the order can differ. Otherwise, model labels are considered.
+        If the task is 'zero-shot-classification', these are the candidate labels.
+        :param batch_size: Number of queries to be processed at a time.
         """
         super().__init__()
-
-        self.devices, _ = initialize_device_settings(use_cuda=use_gpu)
+        devices, _ = initialize_device_settings(use_cuda=use_gpu, multi_gpu=False)
+        device = 0 if devices[0].type == "cuda" else -1
+
+        self.model = pipeline(
+            task=task, model=model_name_or_path, tokenizer=tokenizer, device=device, revision=model_version
+        )
+
+        self.labels = labels
+        if task == "zero-shot-classification":
+            if labels is None:
+                raise ValueError("Candidate labels must be provided for task zero-shot-classification")
+        elif task == "text-classification":
+            labels_from_model = [label for label in self.model.model.config.id2label.values()]
+            if labels is None or set(labels) != set(labels_from_model):
+                self.labels = labels_from_model
+                logger.warning(
+                    f"The labels are not provided or do not match the model labels. Then the model labels are used.\n"
+                    f"Provided labels: {labels}\n"
+                    f"Model labels: {labels_from_model}"
+                )
+        self.task = task
         self.batch_size = batch_size
-        device = 0 if self.devices[0].type == "cuda" else -1
-
-        model = AutoModelForSequenceClassification.from_pretrained(model_name_or_path)
-        tokenizer = AutoTokenizer.from_pretrained(model_name_or_path)
 
-        self.query_classification_pipeline = TextClassificationPipeline(model=model, tokenizer=tokenizer, device=device)
+    def _get_edge_number(self, label):
+        return self.labels.index(label) + 1
```

Comment:
We're about to merge a PR that improves management of nodes with a dynamic number of output edges: https://github.com/deepset-ai/haystack/pull/2850 Can you wait for it to be merged and adapt the code to use the new system? It's going to be merged today or tomorrow anyway

Reply:
This little method (beautifully written by you :smiley:) simply matches a label to a specific output edge.
Does #2850 impact this? How?